common/src/main/kotlin/entity/DiscordDiscoveryMetadata.kt

```diff
@@ -0,0 +1,56 @@
+package dev.kord.common.entity
+
+import dev.kord.common.entity.optional.Optional
+import dev.kord.common.entity.optional.OptionalBoolean
+import dev.kord.common.entity.optional.OptionalSnowflake
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class DiscordDiscoveryMetadata(
+    @SerialName("guild_id")
+    val guildId: Snowflake,
+    val keywords: Optional<List<String>>,
+    @SerialName("emoji_discoverability_enabled")
+    val emojiDiscoverabilityEnabled: Boolean,
+    @SerialName("partner_actioned_timestamp")
+    val partnerActionedTimestamp: Optional<String>,
+    @SerialName("partner_application_timestamp")
+    val partnerApplicationTimestamp: Optional<String>,
+    @SerialName("categories_id")
+    val categoriesIds: List<Snowflake>
+)
+
+@Serializable
+data class DiscordDiscoveryCategory(
+    val id: Int,
+    val name: Name,
+    @SerialName("is_primary")
+    val isPrimary: Boolean
+) {
+    @Serializable
+    data class Name(
+        val default: String,
+        val localizations: Optional<Map<String, String>>
+    )
+}
+
+@Serializable
+data class DiscoveryTermValidationResponse(val valid: Boolean)
```

Comment:
Move this to rest module under json then response.

Edit: As I'm reviewing I realized that these are requests